### PR TITLE
Add mongoDB and a placeholder User model

### PR DESCRIPTION
**Pull request by umbauk**
- Add mongoose package
- Add mongoDB connection to app.js
- Add `predev` script to package.json to start mongod in the background and 'stop' script to shutdown mongo server
- Update README.md with installation, run and stop instructions

### CODE REVIEW

**Review by your-welcome — APPROVED** [review]

**Review by Disquietude — APPROVED** [review]

File: server/app.js [hunk 1]
```diff
@@ -3,10 +3,19 @@ const express = require("express");
 const { join } = require("path");
 const cookieParser = require("cookie-parser");
 const logger = require("morgan");
+const mongoose = require("mongoose");
 
 const indexRouter = require("./routes/index");
 const pingRouter = require("./routes/ping");
 
+// Connect to the database
+const mongoDB = "mongodb://127.0.0.1/mail-sender-dev";
```
Comment (skassam21) [inline server/app.js:12]: This should variable ideally should be stored in a config file, and then the config file should be referencing an environment variable

File: server/app.js [hunk 2]
```diff
@@ -3,10 +3,19 @@ const express = require("express");
...
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true }).then(() => {
+  console.log("Connected to database...");
+});
+const db = mongoose.connection;
+db.on("error", console.error.bind(console, "MongoDB connection error:"));
+
 const { json, urlencoded } = express;
 
 var app = express();
```
Comment (skassam21) [inline server/app.js:21]: Now that we are using const, this should really be const (I know you didnt write this loll)

File: server/models/user.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+// Placeholder User schema
+const UserSchema = new Schema({
+  firstname: { type: String, required: true },
+  lastname: { type: String, required: true },
+  email: { type: String, required: true },
+  password: { type: String, required: true },
+});
```
Comment (skassam21) [inline server/models/user.js:10]: Looks like a good starting point, I'd probably make email unique 
Are we using a linter / prettier on this project yet?

**Review by skassam21 — COMMENTED** [review]
Some comments to consider in the future

**Pull request merged**